packages/simcore-sdk/tests/unit/test_node_ports_common_file_io_utils.py

```diff
@@ -95,6 +98,31 @@ async def test_check_for_aws_http_errors(
             assert _check_for_aws_http_errors(exception) is test_params.will_retry
 
 
+async def test_upload_file_to_presigned_links_raises_aws_s3_400_request_time_out_error(
+    mocker: MockerFixture,
+    create_upload_links: Callable[[int, ByteSize], Awaitable[FileUploadSchema]],
+    create_file_of_size: Callable[[ByteSize], Path],
+):
+    file_size = ByteSize(1)
+    upload_links = await create_upload_links(1, file_size)
+
+    mocker.patch(
+        "simcore_sdk.node_ports_common.file_io_utils._upload_file_part",
+        side_effect=AWSS3400RequestTimeOutError(body="nothing"),
+    )
+
+    async with ProgressBarData(steps=1) as progress_bar:
+        with pytest.raises(AWSS3400RequestTimeOutError):
+            await upload_file_to_presigned_links(
+                session=AsyncMock(),
+                file_upload_links=upload_links,
+                file_to_upload=create_file_of_size(file_size),
+                num_retries=0,
+                io_log_redirect_cb=None,
+                progress_bar=progress_bar,
```

Comment:
Here I don't see any test about the complete function with the retrial

Reply:
What I tried to have is the following:

- Upon error, the upload is abandoned as is and storage will clean up the link. This is similar to what happened before during an error.
- Start again the upload and it requests new multipart upload links.

Reply:
As discussed I've wrapped the retry on the entire upload function.